concordium-consensus/tools/database-exporter/Main.hs

```diff
@@ -47,7 +47,8 @@ main = do
     Identity conf <- execParser opts
     case conf of
         Check file -> checkDatabase file
-        Export db filepath -> exportDatabaseV3 db filepath
+        Export db filepath n | n > 0 -> exportDatabaseV3 db filepath n
+        Export _ _ _ -> putStrLn "Chunk size should be larger than 0"
```

Comment:
This strictly speaking breaks compatibility. Our scripts for export will no longer work, nor is this compatible with existing nodes directly.

We can keep using the existing exporter, or presumably use `--chunk-size=2^63` to achieve this effect (but the file names will be different) but regardless, this needs to be planned so that we maintain out of band catchup also for existing 4.2.* nodes that have not upgraded immediately.

Reply:
I brought it up once or twice that removing the existing possibility to export the complete database without specifying a chunk size might break compatibility, and I was collectively assured it won't be an issue.

Reply:
As you say, we can continue to use the existing database exporter. However, I think we would expect to deprecate the old exports once the current version of the node supports the new out-of-band catch-up. If someone needs to do out-of-band catch-up, it is because they are starting a node, in which case they should be starting the latest version.

Reply:
@mkmks I am not sure where you brought this up, I am not aware of it. In this case it is also very easy to be backwards compatible.

While we can discuss amongst ourselves here, this needs to be coordinated with how the tools are used. So I expect a note to be written about this and circulated to at least Felice so that they may plan around it.

And we cannot expect that everybody "should" be starting the latest node, but that might not be the reality. People might be starting older nodes, since they don't have to upgrade, and the upgrade might require extra work for them.

In any case, I expect an explicit written plan on how migration is meant to work. If that already exists somewhere point me to it.